apps/prairielearn/python/python_helper_sympy.py

```diff
@@ -297,7 +309,9 @@ def ast_check(expr: str, locals_for_eval: LocalsForEval) -> None:
     CheckFunctions(locals_for_eval["functions"]).visit(root)
 
     # Disallow variables that are not in locals_for_eval
-    CheckVariables(locals_for_eval["variables"]).visit(root)
+    CheckVariables(locals_for_eval["variables"], locals_for_eval["functions"]).visit(
```

Comment:
For future reference: After looking at this, I think these `Check...` classes can actually all be condensed into a single class that subclasses `NodeVisitor`.